### ISSUE TITLE
Checkov failing to perform a check against a Terraform Plan when skipping a specific AWS check

### ISSUE THREAD

**Issue by citrix-eg** [bridgecrewio/checkov#3744]
**Describe the issue**
Checkov failing to perform a check against a Terraform Plan when skipping a specific AWS check

**Examples**
```
resource "aws_iam_role" "backend" {
  #checkov:skip=CKV_AWS_274:TODO Generate a policy from CloudTrail later
  name                = "${var.repo}-foo"
  assume_role_policy  = data.aws_iam_policy_document.backend-assume.json
  managed_policy_arns = ["arn:aws:iam::aws:policy/AdministratorAccess"]
  inline_policy {
    name   = "foo"
    policy = data.aws_iam_policy_document.backend-permissions.json
  }
  tags = {
    component = var.foo
  }
}
```

**Exception Trace**
```
Traceback (most recent call last):
  File "/usr/local/bin/checkov", line 9, in <module>
    sys.exit(run())
  File "/usr/local/lib/python3.9/site-packages/checkov/main.py", line 355, in run
    scan_reports = runner_registry.run(external_checks_dir=external_checks_dir, files=config.file,
  File "/usr/local/lib/python3.9/site-packages/checkov/common/runners/runner_registry.py", line 79, in run
    self.runners[0].run(root_folder, external_checks_dir=external_checks_dir, files=files,
  File "/usr/local/lib/python3.9/site-packages/checkov/terraform/plan_runner.py", line 81, in run
    self.check_tf_definition(report, root_folder, runner_filter)
  File "/usr/local/lib/python3.9/site-packages/checkov/terraform/plan_runner.py", line 99, in check_tf_definition
    self.run_block(definition[block_type], None, full_file_path, root_folder, report, scanned_file,
  File "/usr/local/lib/python3.9/site-packages/checkov/terraform/plan_runner.py", line 119, in run_block
    results = registry.scan(scanned_file, entity, [], runner_filter, report_type=CheckType.TERRAFORM_PLAN)
  File "/usr/local/lib/python3.9/site-packages/checkov/common/checks/base_check_registry.py", line 127, in scan
    result = self.run_check(check, entity_configuration, entity_name, entity_type, scanned_file, skip_info)
  File "/usr/local/lib/python3.9/site-packages/checkov/common/checks/base_check_registry.py", line 141, in run_check
    result = check.run(
  File "/usr/local/lib/python3.9/site-packages/checkov/common/checks/base_check.py", line 70, in run
    check_result["result"] = self.scan_entity_conf(entity_configuration, entity_type)
  File "/usr/local/lib/python3.9/site-packages/checkov/terraform/checks/resource/base_resource_check.py", line 43, in scan_entity_conf
    return self.scan_resource_conf(conf)
  File "/usr/local/lib/python3.9/site-packages/checkov/terraform/checks/resource/aws/IAMManagedAdminPolicy.py", line 42, in scan_resource_conf
    if conf.get("policy_arn")[0] == ADMIN_POLICY_ARN:
TypeError: 'NoneType' object is not subscriptable
```

**Desktop (please complete the following information):**
 - MacOS 11.7
 - Checkov Version 2.2.0

**Additional context**
This fails as of whateverv version CKV_AWS_274 was added. Last time a build didn't crash I was using 2.1.294 and it worked.
Also if I skip it with a command-line switch then this crash does not happen (which is going to be my temp workaround)

**Comment by gruebel** [comment]
hey @citrix-eg thanks for reaching out. Sadly I can't reproduce it. The error also indicates it is not the provided `aws_iam_role`, which causes the error, there has to be an IAM attachment resource like `aws_iam_policy_attachment` in your plan file.

**Comment by citrix-eg** [comment]
I can pass you a full plan file although I'd prefer to do it out-of-band from here. I tried pretty hard to scrub it of sensitive stuff but would still like to minimize risk. Can I pass it to you via Slack or something?

Its worth noting that this issue does not occur if I target that specific resource so that does kind of jive with what you said that its not the resource *itself* but something in the plan causing issues.

There are no attachment resources that I can see in either the raw TF or in the plan JSON but again--would be happy to kick that to you OOB

**Comment by citrix-eg** [comment]
Gah, shame on me! I clipped off too much of the error itself and only pasted the traceback! Yes, this is not even remotely related to the role I mentioned at first. The fact that I was responding to the initial checkov flagging for that resource threw me off.

Here is shell output including invocation but not including the traceback I sent (which is the same):
```
➜ checkov -f ~/Downloads/tfplan.json --repo-root-for-plan-enrichment . --download-external-modules true --framework terraform_plan --check CKV_AWS_274 
2022-10-27 08:50:43,940 [MainThread  ] [ERROR]  Failed to run check:
Disallow IAM roles, users, and groups from using the AWS AdministratorAccess policy for configuration:
{'start_line': [0], 'end_line': [0], '__address__':
'module.vpc.aws_iam_role_policy_attachment.vpc_flow_log_cloudwatch[0]', '__change_actions__':
['create']} at file: /../../../../Downloads/tfplan.json
```

That helps because it makes my plan file a lot smaller and I feel a lot more confident I sanitized it well so I can post it here. I generated this plan by targeting the specific resource that was mentioned in the error message so hopefully this is helpful!

Some additional context, this is generated from the [common VPC module](https://registry.terraform.io/modules/terraform-aws-modules/vpc/aws/latest).

[tfplan.json.zip](https://github.com/bridgecrewio/checkov/files/9879677/tfplan.json.zip)
